blocks/feed-newsroom/feed-newsroom.js

```diff
@@ -191,13 +195,36 @@ export default async function decorate(block) {
 </form>`;
   const uniqYears = Array.from(new Set(results.map((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return parseInt(itsDate[itsDate.length - 1], 10); })));
   // eslint-disable-next-line
-  const yroptions = uniqYears.reduce((accum, current) => { accum += "<option value='"+current+"'>" + current + "</option>"; return accum;},"");
-  filterDiv.querySelector('select').innerHTML = filterDiv.querySelector('select').innerHTML + yroptions;
+  const yroptions = uniqYears.reduce((accum, current) => { accum += "<option value='" + current + "'>" + current + "</option>"; return accum; }, "");
+  filterDiv.querySelector('#news_year').innerHTML = filterDiv.querySelector('#news_year').innerHTML + yroptions;
+  const categoryDetails = await fetchTagsOrCategories('', 'categories', 'newsroom', locale);
+  // eslint-disable-next-line
+  const categoryOptions = categoryDetails.reduce((accum, current) => { accum += "<option value='" + current.id + "'>" + current.name + "</option>"; return accum; }, "");
+  filterDiv.querySelector('#news_category').innerHTML = filterDiv.querySelector('#news_category').innerHTML + categoryOptions;
 
   filterDiv.querySelector('form .filter-nav button').addEventListener('click', () => {
-    const searchYear = Number(filterDiv.querySelector('form .filter-nav select').value);
-    searchResults = results.filter((x) => { const itsDate = getFormattedDate(new Date(parseInt(x[blockCfg.sort.trim().toLowerCase()], 10))).split(', '); return (parseInt(itsDate[itsDate.length - 1], 10) === searchYear); });
-    loadYearResults(block, blockType, searchResults, blockCfg, locale);
+    const searchYear = Number(filterDiv.querySelector('form .filter-nav #news_year').value);
+    const searchCategory = filterDiv.querySelector('form .filter-nav #news_category').value;
+    let filteredResults = window.jslinq(results);
+    if (searchYear) {
+      filteredResults = filteredResults.where((el) => {
+        const itsDate = getFormattedDate(new Date(parseInt(el[blockCfg.sort.trim().toLowerCase()], 10))).split(', ');
```

Comment:
we have timestamp in the query-index then year can be easily detected using below code
const date = new Date(timestamp);
const year = date.getFullYear();

not sure if i am missing something

Reply:
Sure, just would like to keep consistency between codes for feed.js as shown below & feed-newsroom.js leveraging - getFormattedDate rather to get the year. 

if (fieldName === 'publisheddate') {
            div.classList.add('date');
            div.textContent = getFormattedDate(new Date(parseInt(result[fieldName], 10)));